### DISCUSSION TITLE
Using global filters on non-hidden columns only

### DISCUSSION THREAD

**Discussion by rajat-sr** [TanStack/table#2373] [General]
I am using `setHiddenColumns` to hide certain columns in a table. The _hidden_ columns can be made visible by selecting it from a list of columns.
![image](https://user-images.githubusercontent.com/13731210/82950542-19e37e80-9fc3-11ea-827f-b6069eff0d4d.png)

There's also a search bar at the top of the table which is used to filter the rows based on the text entered. I am using properties from `useGlobalFilters` hook to implement the search functionality.
The `setGlobalFilter` searches/filters on every column in the table (which is fine given its _global_ nature). There is another hook called `useFilters` which provides properties to add column specific filters.

But I am looking for something in-between. I want a global search/filter that only filters on the visible columns (and excludes hidden columns from its search database). I cannot do this using `setFilter` method from `useFilters` hook since it only accepts one column id.

Is there another function existing that allows using global filters on non-hidden columns? If there is no such existing function, is there a way I can implement it?

**Comment by joshruien** [comment]
Did you manage to solve this issue?

**Reply by rajat-sr** [reply]
No. Do you have a solution?

**Reply by joshruien** [reply]
Unfortunately not... But i came across an example where he was able to customise the global filter. https://codesandbox.io/s/specific-columns-global-filter-n1k4v?file=/src/Table.tsx:1598-1747

I tried working around with this but to no avail.

**Comment by TjarkTV** [comment]
For anyone wondering you can toggle if the column is filterable with 
`column.columnDef.enableGlobalFilter = !column.getCanGlobalFilter();`
just put it inside your function that toggles the visibility
